examples/trl/sft.py

```diff
@@ -56,131 +73,151 @@ class ScriptArguments:
+            else:
+                total_tokens += len(tokenizer.tokenize(text))
+
+        return total_characters / total_tokens
+
+    def prepare_sample_text(example):
+        """Prepare the text from a sample of the dataset."""
+        text = f"Question: {example['question']}\n\nAnswer: {example['response_j']}"
+        return text
+
+    def create_datasets(tokenizer, args, seed=None):
+        if args.dataset_name:
+            dataset = load_dataset(
+                args.dataset_name,
+                data_dir=args.subset,
+                split=args.split,
+                token=script_args.token,
+                num_proc=args.num_workers if not args.streaming else None,
+                streaming=args.streaming,
+            )
+        elif args.dataset_path:
+            ext = args.dataset_path.split(".")[-1]
+            dataset = load_dataset(ext, data_files=args.dataset_path, split=args.split, token=script_args.token)
         else:
-            total_tokens += len(tokenizer.tokenize(text))
-
-    return total_characters / total_tokens
-
+            logger.info("No dataset_name or dataset_path")
+            exit()
+        if not args.dataset_path and args.streaming:
+            logger.info("Loading the dataset in streaming mode")
+            valid_data = dataset.take(args.size_valid_set)
+            train_data = dataset.skip(args.size_valid_set)
+            train_data = train_data.shuffle(buffer_size=args.shuffle_buffer, seed=seed)
+        else:
+            dataset = dataset.train_test_split(test_size=args.validation_split_percentage * 0.01, seed=seed)
+            train_data = dataset["train"]
+            valid_data = dataset["test"]
+            logger.info(f"Size of the train set: {len(train_data)}. Size of the validation set: {len(valid_data)}")
+        if args.dataset_name == "lvwerra/stack-exchange-paired":
+            chars_per_token = chars_token_ratio(train_data, tokenizer)
+            logger.info(f"The character to token ratio of the dataset is: {chars_per_token:.2f}")
+            formating_func = prepare_sample_text
+        else:
+            formating_func = None
+        return train_data, valid_data, formating_func
 
-def prepare_sample_text(example):
-    """Prepare the text from a sample of the dataset."""
-    text = f"Question: {example['question']}\n\nAnswer: {example['response_j']}"
-    return text
+    low_cpu_mem_usage = True
+    if is_deepspeed_available():
+        from transformers.integrations.deepspeed import is_deepspeed_zero3_enabled
 
+        if is_deepspeed_zero3_enabled():
+            low_cpu_mem_usage = False
 
-def create_datasets(tokenizer, args, seed=None):
-    dataset = load_dataset(
-        args.dataset_name,
-        data_dir=args.subset,
-        split=args.split,
+    base_model = AutoModelForCausalLM.from_pretrained(
+        script_args.model_name_or_path,
+        low_cpu_mem_usage=low_cpu_mem_usage,
+        torch_dtype=torch.bfloat16,
```

Comment:
Checking on this @libinta : Do you want/need to force `torch.bfloat16` for the model?

Reply:
yes